### PR TITLE
Bug 1858387, clarify how to update the console URL

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1858387

Preview Builds:
http://file.rdu.redhat.com/~ahardin/12-11-2020/change-console-url/web_console/customizing-the-web-console.html#customizing-the-web-console-url_customizing-web-console

http://file.rdu.redhat.com/~ahardin/12-11-2020/change-console-url/web_console/configuring-web-console.html#web-console-configuration_configuring-web-console

Functionality for 4.5+

### CODE REVIEW

**Review by spadgett — COMMENTED** [review]
We'll also want to cover the case where the user specifies a custom cert, which requires creating a secret in the openshift-config namespace that has the key and certificate.

For example,

```
$ oc create secret tls console-tls --key=key.pem --cert=cert.pem -n openshift-config
```

Then you'll edit to the config resource with the following:

```
spec:
  route:
    hostname: console.example.com
    secret:
      name: console-tls
```

cc @jhadvig

**Comment by ahardin-rh** [comment]
@spadgett Thanks! We can cross-check what is on this existing page and add any missing procedures: https://docs.openshift.com/container-platform/4.5/web_console/customizing-the-web-console.html#creating-custom-links_customizing-web-console

**Comment by ahardin-rh** [comment]
@spadgett @jhadvig I made some updates PTAL.

Updated Preview Builds:
http://file.rdu.redhat.com/~ahardin/12-11-2020/change-console-url/web_console/customizing-the-web-console.html#customizing-the-web-console-url_customizing-web-console

http://file.rdu.redhat.com/~ahardin/12-11-2020/change-console-url/web_console/configuring-web-console.html#web-console-configuration_configuring-web-console

**Comment by ahardin-rh** [comment]
@yapei Please provide QE review. Thank you!

**Comment by yapei** [comment]
look good to me

**Review by yapei — APPROVED** [review]
/lgtm

File: modules/web-console-configuration.adoc [hunk 1]
```diff
@@ -29,10 +28,5 @@ spec:
...
   consoleURL: "" <2>
 ----
-<1> Specify the URL of the page to load when a user logs out of the web console.
-If you do not specify a value, the user returns to the login page for the
-web console. Specifying a `logoutRedirect` URL allows your users to perform
-single logout (SLO) through the identity provider to destroy their single
-sign-on session.
-<2> The web console URL. You cannot modify this parameter value. If you do, the
-cluster resets it to the default value.
+<1> Specify the URL of the page to load when a user logs out of the web console.If you do not specify a value, the user returns to the login page for the web console. Specifying a `logoutRedirect` URL allows your users to perform single logout (SLO) through the identity provider to destroy their single sign-on session.
```
Comment (spadgett) [inline modules/web-console-configuration.adoc:31]: Missing a space after the period

**Review by spadgett — APPROVED** [review]
lgtm

**Review by jhadvig — APPROVED** [review]

File: modules/customizing-the-web-console-URL.adoc [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// Module included in the following assemblies:
+//
+// * web_console/customizing-the-web-console.adoc
+
+[id="customizing-the-web-console-url_{context}"]
+= Customizing the web console URL
+
+To update the web console URL, `consoleURL`, to a custom value:
```
Comment (kalexand-rh) [inline modules/customizing-the-web-console-URL.adoc:8]:
```suggestion
You can update the web console URL, `consoleURL`, to a custom value.

.Procedure
```
You need the `.Procedure` heading here.

File: modules/customizing-the-web-console-URL.adoc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+. Modify the cluster instance created by default during installation in the `consoles.operator.openshift.io` custom resource:
++
+[source,terminal]
+----
+$ oc patch consoles.operator.openshift.io cluster --patch '{"spec":{"route":{"hostname":"console.example.com"}}}' --type=merge
+----
+
+. (Optional) If you specify a custom certificate, you must create a secret in the `openshift-config` namespace that has the key and certificate. For example:
```
Comment (kalexand-rh) [inline modules/customizing-the-web-console-URL.adoc:17]:
```suggestion
. If you specify a custom certificate, you must create a secret in the `openshift-config` namespace that has the key and certificate. For example:
```
It looks like this step is required in a certain context, which you've explained well, so it's not really optional.

File: modules/customizing-the-web-console-URL.adoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
++
+[source,terminal]
+----
+$ oc create secret tls console-tls --key=key.pem --cert=cert.pem -n openshift-config
+----
++
+Then, edit the configuration resource with the following:
```
Comment (kalexand-rh) [inline modules/customizing-the-web-console-URL.adoc:24]:
```suggestion
Then, add the following stanza to the configuration resource:
```

File: modules/customizing-the-web-console-URL.adoc [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+[id="customizing-the-web-console-url_{context}"]
+= Customizing the web console URL
+
+To update the web console URL, `consoleURL`, to a custom value:
+
+. Modify the cluster instance created by default during installation in the `consoles.operator.openshift.io` custom resource:
++
+[source,terminal]
+----
+$ oc patch consoles.operator.openshift.io cluster --patch '{"spec":{"route":{"hostname":"console.example.com"}}}' --type=merge
```
Comment (kalexand-rh) [inline modules/customizing-the-web-console-URL.adoc:14]: Do you just change `console.example.com` to whatever you want the URL to be? I think this is worth making explicit.

File: modules/web-console-configuration.adoc [hunk 6]
```diff
@@ -29,10 +28,5 @@ spec:
...
 ----
-<1> Specify the URL of the page to load when a user logs out of the web console.
-If you do not specify a value, the user returns to the login page for the
-web console. Specifying a `logoutRedirect` URL allows your users to perform
-single logout (SLO) through the identity provider to destroy their single
-sign-on session.
-<2> The web console URL. You cannot modify this parameter value. If you do, the
-cluster resets it to the default value.
+<1> Specify the URL of the page to load when a user logs out of the web console. If you do not specify a value, the user returns to the login page for the web console. Specifying a `logoutRedirect` URL allows your users to perform single logout (SLO) through the identity provider to destroy their single sign-on session.
+<2> The web console URL. To update this to a custom value, see Customizing the web console in {product-title}.
```
Comment (kalexand-rh) [inline modules/web-console-configuration.adoc:32]: The title here doesn't match the module title. I also think that this needs to be in bold, but I can't find a ruling in our doc guidelines.

**Review by kalexand-rh — COMMENTED** [review]
I have a few questions and suggestions, but this is looking good!

**Pull request merged**